### PR TITLE
feat: entity batch stream

### LINKED ISSUE #363: Add "batch" operation to Entity stream
It should be possible to enable batch behavior on entities. Being forced to use per-item concurrency is not acceptable in many stream use-cases, like replicating data to another data base. It is important to use batch write operations for performance.

### PR DESCRIPTION
**Pull request by thantos**
Closes #363

### CODE REVIEW

File: apps/tests/aws-runtime/test/test-service.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ import {
   transaction,
   workflow,
 } from "@eventual/core";
-import type openapi from "openapi3-ts";
+import openapi from "openapi3-ts";
```
Comment (sam-goodwin) [inline apps/tests/aws-runtime/test/test-service.ts:35]: Regression?

File: apps/tests/aws-runtime/test/test-service.ts [hunk 2]
```diff
@@ -616,20 +617,24 @@ export const counterWatcher = counter.stream(
...
-    await counter.set({
-      namespace: "default",
-      id: value!.id,
-      n: (value?.n ?? 0) + 1,
-      optional: undefined,
-    });
-    console.log("send signal to", value!.id);
-    await entitySignal.sendSignal(value!.id);
+  async (items) => {
+    await Promise.all(
```
Comment (sam-goodwin) [inline apps/tests/aws-runtime/test/test-service.ts:624]: Do we support returning partial failures?

File: packages/@eventual/core-runtime/src/handlers/entity-stream-worker.ts [hunk 3]
```diff
@@ -18,19 +24,44 @@ export function createEntityStreamWorker(
...
+
+        return {
+          failedItemIds: [
+            ...results.rejected.map(([e]) => e.id),
+            ...results.fulfilled
+              .filter(([, r]) => r === false)
+              .map(([e]) => e.id),
+          ],
+        };
+      }
```
Comment (sam-goodwin) [inline packages/@eventual/core-runtime/src/handlers/entity-stream-worker.ts:65]: Cool, so this supports partial successful items?

Comment (thantos) [reply packages/@eventual/core-runtime/src/handlers/entity-stream-worker.ts:65]: Yeah!

The single item handler you can return false or throw for a partial failure (this was there before).

The batch API support returning the IDs to "fail".

File: apps/tests/aws-runtime/test/test-service.ts (hunk 2 above)
Comment (thantos) [reply apps/tests/aws-runtime/test/test-service.ts:624]: We do, see below. The batch API needs to handle partial failures manually using the item.id. The per item API you just need to return false.

**Review by sam-goodwin — APPROVED** [review]

**Pull request merged**